### ISSUE TITLE
Polkit auth dialog action messages are not localized properly

### ISSUE THREAD

**Issue by rgeorgiev583** [systemd/systemd#21551]
**systemd version the issue has been seen with**
 > systemd 249 (249.7-1-arch)

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > `Linux pavilion 5.10.82-1-lts #1 SMP Fri, 26 Nov 2021 10:51:51 +0000 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > The action message in the authentication dialog should be displayed in the language defined by the value of the `LC_MESSAGES` variable in `/etc/locale.conf` on my system (which is `en_US.UTF-8`).

**Unexpected behaviour you saw**
 > The action message was displayed in the language defined by the `LANG` variable instead (`bg_BG.UTF-8`).

**Steps to reproduce the problem**
 > 1. Attempt to execute a unit-related job (e.g. `systemctl stop preload`) without explicitly granting privileges to `systemctl`.
 > 2. Observe the polkit agent authentication dialog appear.

I expect that systemd would configure the proper system locale so that gettext would obey the value of `LC_MESSAGES` which is defined in `/etc/locale.conf` for all kinds of messages.  This is the expected behavior as per `man 7 locale`.

**Additional program output to the terminal or log subsystem illustrating the issue**

![systemd-stop-service-polkit-auth-dialog](https://user-images.githubusercontent.com/5637095/143779617-f2526c9b-11b4-498d-a7da-c4f0b53a6c6d.png)

**Comment by rgeorgiev583** [comment]
I grepped for the message which appeared on my screen in the code and discovered from the `bg.po` file that gettext is invoked for this functionality in `src/core/dbus-unit.c`.  It seems that the `N_` macro is used there to obtain the translated string, which seemed a bit odd to me (I hadn't seen such a macro being used in gettext-based software before).

The definition of said macro is in `src/basic/locale-util.h`, and it seems just plain wrong to me:
```c
#define N_(String) String
```

At the same time, the `_` macro (which is normally used to translate strings) is defined as follows (which seems the proper way to do it to me):
```c
#define _(String) gettext(String)
```

I think the `_` macro should be used instead of `N_`.

Moreover, I grepped for `setlocale`, and discovered that there is a function which is supposed to initialize the gettext context (`init_gettext`) which invokes `setlocale` but never gets invoked itself.  It is also unclear to me when and how systemd actually reads the value of `/etc/locale.conf`.

**Comment by yuwata** [comment]
Hm? The message is really generated by systemd? I cannot find "An application is attempting to perform an action that requires privileges. Authentication is required to perform this action." in our source.

**Comment by rgeorgiev583** [comment]
> Hm? The message is really generated by systemd? I cannot find "An application is attempting to perform an action that requires privileges. Authentication is required to perform this action." in our source.

No, I'm not talking about the message you quoted (which is located in the `polkit-kde-agent-1` source, indeed).   I'm talking about the message which is translated into Bulgarian on the screenshot (which is the Polkit action message).  The gettext msgid for this message (as well all other job-related ones) is referenced in `src/core/dbus-unit.c`, in the definition of the `polkit_message_for_job` array:
```c
static const char *const polkit_message_for_job[_JOB_TYPE_MAX] = {
        [JOB_START]       = N_("Authentication is required to start '$(unit)'."),
        [JOB_STOP]        = N_("Authentication is required to stop '$(unit)'."),
        [JOB_RELOAD]      = N_("Authentication is required to reload '$(unit)'."),
        [JOB_RESTART]     = N_("Authentication is required to restart '$(unit)'."),
        [JOB_TRY_RESTART] = N_("Authentication is required to restart '$(unit)'."),
};
```

What is problematic here (IMO) is that `N_` is defined as follows:
```c
#define N_(String) String
```
which does not mention `gettext` at all (unlike the definition of the `_` macro).

**Comment by rgeorgiev583** [comment]
BTW the source file lines mentioned in the `bg.po` file also seem wrong:
```po
#: src/core/dbus-unit.c:359
msgid "Authentication is required to start '$(unit)'."
msgstr "За стартиране на „$(unit)“ е необходима идентификация."

#: src/core/dbus-unit.c:360
msgid "Authentication is required to stop '$(unit)'."
msgstr "За спиране на „$(unit)“ е необходима идентификация."

#: src/core/dbus-unit.c:361
msgid "Authentication is required to reload '$(unit)'."
msgstr "За презареждане на „$(unit)“ е необходима идентификация."

#: src/core/dbus-unit.c:362 src/core/dbus-unit.c:363
msgid "Authentication is required to restart '$(unit)'."
msgstr "За рестартиране на „$(unit)“ е необходима идентификация."
```
I'm not certain how much of a problem is this but it needs to be fixed as well (I suppose there is some tool which generates this metadata, no?).

**Comment by poettering** [comment]
N_() is for marking strings for translation so that xgettext finds them. It has no effect other than that. (At least that's how I remember this to work.) Hence, the fact that this is mapped to a NOOP macro is intended.

I think it's polkit which use the wrong locale settings but i honestly don't know how polkit l10n works...
